test/extensions/filters/http/transformation/inja_transformer_test.cc

```diff
@@ -227,6 +227,23 @@ TEST(Extraction, ExtractIdFromHeader) {
   EXPECT_EQ("123", res);
 }
 
+TEST(Extraction, ExtractIdFromBody) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"},
+                                         {":authority", "www.solo.io"}};
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  extractor.set_regex("id: ([0-9]{4})");
+  extractor.set_subgroup(1);
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body(R"EOF(
+    id: 1234
```

Comment:
`regex_match` would not work in this case, and this is a very realistic case where we don't want to make the regex match the whole body, but just a substring of it.